rotkehlchen/exchanges/kucoin.py

```diff
@@ -0,0 +1,774 @@
+                        error=str(e),
+                        raw_result=raw_result,
+                    )
+                    if isinstance(e, (UnknownAsset, UnsupportedAsset)):
+                        asset_tag = 'unknown' if isinstance(e, UnknownAsset) else 'unsupported'
+                        error_msg = f'Found {asset_tag} kucoin asset {e.asset_name}'
+
+                    self.msg_aggregator.add_error(
+                        f'Failed to deserialize a kucoin {case} result. {error_msg}. Ignoring it. '
+                        f'Check logs for more details')
+                    continue
+
+                if result is None and skip_reason is not None:
+                    if skip_reason == SkipReason.BEFORE_TIMESTAMP_RANGE:
+                        log.debug(f'Found a kucoin {case} {skip_reason}. Stop requesting.')
+                        break
+
+                    if skip_reason in (
+                        SkipReason.AFTER_TIMESTAMP_RANGE,
+                        SkipReason.INNER_MOVEMENT,
+                    ):
+                        log.debug(f'Found a kucoin {case} {skip_reason}. Skipping it.')
+                        continue
+
+                    raise AssertionError(f'Unexpected skip reason: {skip_reason}. Update conditions')  # noqa: E501
+
+                assert result is not None
+                results.append(result)  # type: ignore # type is known
+
+            # Stop requesting, either when a result was before the range or
+            # totalPage indicates to stop paginating
+            if is_before_start_ts is True or total_page in (0, current_page):
+                break
+
+            # Update pagination params per endpoint
+            # NB: Copying the dict before updating it prevents losing the call args values
+            call_options = call_options.copy()
+            call_options['currentPage'] = current_page + 1
+
+        return results
+
+    def _deserialize_accounts_balances(
+            self,
+            response_dict: Dict[str, List[Dict[str, Any]]],
+    ) -> Dict[Asset, Balance]:
+        """May raise RemoteError
+        """
+        try:
+            accounts_data = response_dict['data']
+        except KeyError as e:
+            msg = 'Kucoin balances JSON response is missing data key'
+            log.error(msg, response_dict)
+            raise RemoteError(msg) from e
+
+        assets_balance: DefaultDict[Asset, Balance] = defaultdict(Balance)
+        for raw_result in accounts_data:
+            try:
+                amount = FVal(raw_result['balance'])
+                if amount == ZERO:
+                    continue
+
+                asset_symbol = raw_result['currency']
+            except (KeyError, ValueError) as e:
+                msg = str(e)
+                if isinstance(e, KeyError):
+                    msg = f'Missing key in account: {msg}.'
+
+                log.error(
+                    'Failed to deserialize a kucoin account',
```

Comment:
here and below I suppose more appropriate would be to swap account with balance, right?

Reply:
Fine, yes. I used the account word as each object comes from an account (from the accounts endpoint). But for rotki it makes sense to use the word balance.